src/pokemon.c

```diff
@@ -8763,16 +8764,22 @@ void HandleSetPokedexFlag(u16 nationalNum, u8 caseId, u32 personality)
 
 const u8 *GetTrainerClassNameFromId(u16 trainerId)
 {
-    if (trainerId >= TRAINERS_COUNT)
-        trainerId = TRAINER_NONE;
-    return gTrainerClassNames[gTrainers[trainerId].trainerClass];
+    if (trainerId >= MAX_TRAINERS_COUNT && trainerId < PARTNERS_COUNT)
```

Comment:
I forgot this in the previous review, sorry :D

Here you check against `MAX_TRAINERS_COUNT` while you check against `TRAINER_PARTNER(PARTNER_NONE))` somewhere else. I think the latter is preferable everywhere to indicate we're looking at partners

Reply:
Correct!